mono/mini/mini-native-types.c

```diff
@@ -95,6 +95,22 @@ typedef struct {
 
 #endif
 
+gsize
+mini_magic_type_size (MonoCompile *cfg, MonoType *type)
```

Comment:
I realize you are just moving code, but a comment near the word "magic" sure might be nice.
What is "magic"?